core/models/instance.py

```diff
@@ -241,10 +254,11 @@ def update_history(
             # logger.info("status_name matches last history:%s " %
             #        last_history.status.name)
             return (False, last_history)
+        status = self.esh_status()
```

Comment:
does it make sense to pull this line up to Line 225 with `activity = self.esh_activity()`?